rakelib/default_plugins.rb

```diff
@@ -168,5 +17,24 @@ def self.is_released?(plugin)
       require 'gems'
       Gems.info(plugin) != "This rubygem could not be found."
     end
+
+    def self.fetch_plugins_for(clazz)
+      LogStash::Json.load(::File.read("rakelib/plugins-metadata.json")).select do |_, metadata|
+        metadata[clazz]
+      end.keys
+    end
```

Comment:
works for me, will update.

- purbon

On Thu, Apr 14, 2016 at 3:05 PM Pier-Hugues Pellerin <
notifications@github.com> wrote:

> In rakelib/default_plugins.rb
> <https://github.com/elastic/logstash/pull/5096#discussion_r59712734>:
>
> > @@ -168,5 +17,24 @@ def self.is_released?(plugin)
> >        require 'gems'
> >        Gems.info(plugin) != "This rubygem could not be found."
> >      end
> > +
> > +    def self.fetch_plugins_for(clazz)
> > +      LogStash::Json.load(::File.read("rakelib/plugins-metadata.json")).select do |_, metadata|
> > +        metadata[clazz]
> > +      end.keys
> > +    end
>
> Its not really a clazz, more like a type or field?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/elastic/logstash/pull/5096/files/c9db449808fbb9aeb4160e3b1d1fa3f6e709aabd#r59712734>
>